analysis_driver/quality_control/median_coverage.py

```diff
@@ -0,0 +1,74 @@
+import os
+from analysis_driver import executor
+from analysis_driver.config import default as cfg
+from analysis_driver.notification import default as ntf
+from .quality_control_base import QualityControl
+
+
+class SamtoolsDepth(QualityControl):
+    def __init__(self, dataset, working_dir, bam_file):
+        super().__init__(dataset, working_dir)
+        self.bam_file = bam_file
+        self.working_dir =  working_dir
+        self.exception = None
+
+
+    def _get_samtools_depth_command(self):
+        samtools_bin = cfg['tools']['samtools']
+        samtools_depth_out_file = ((self.bam_file).rstrip('bam') + 'depth')
+        samtools_depth_command = "%s depth -a -a -q 0 -Q 0 %s > %s" % (samtools_bin, self.bam_file, samtools_depth_out_file)
+        return samtools_depth_command, samtools_depth_out_file
+
+
+
+    def _run_samtools_depth(self):
+        """
+        :return string: the expected outfile from samtools depth
+        """
+        samtools_depth_command, samtools_depth_out_file = self._get_samtools_depth_command()
+        ntf.start_stage('run_samtools_depth')
+        samtools_depth_executor = executor.execute(
+            [samtools_depth_command],
+            job_name='samtoolsdepth',
+            working_dir=self.working_dir,
+            cpus=2,
```

Comment:
samtools is single-threaded bring dow the number of cpu to 1 and mem probably to 6